### PR TITLE
LPS-132761 active view API using REST Builder

### PR DESCRIPTION
**Pull request by nhpatt**
Active view API (storing and recovering JSON) based on REST Builder. Had to modify REST Builder slightly to support APIs without schemas (was supported but I broke it with the batch stuff).

I copied the API as it is but we have to consider if `plid` and `portletId` are required parameters... they should be in the path instead of query parameters.

### CODE REVIEW

**Comment by nhpatt** [comment]
> Please only forward critical changes to Brian Chan during stabilization. Nonurgent changes should wait until the ongoing 7.4 DXP EP2 & CE GA2 release has been completed. For more details on the release timeline and status, see product-delivery.

Oh, neat. But wrong time to post it :P

File: modules/apps/frontend-view-state/frontend-view-state-rest-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                            type: string
+                    application/xml:
+                        schema:
+                            type: string
+            responses:
+                200:
+                    content:
+                        application/json:
+                            schema:
+                                type: object
```
Comment (javierdearcos) [inline modules/apps/frontend-view-state/frontend-view-state-rest-impl/rest-openapi.yaml:67]: I guess it is not possible to model this object, but I have to ask

**Review by javierdearcos — APPROVED** [review]

File: modules/apps/frontend-view-state/frontend-view-state-rest-impl/rest-openapi.yaml (hunk 1 above)
Comment (nhpatt) [reply modules/apps/frontend-view-state/frontend-view-state-rest-impl/rest-openapi.yaml:67]: It's an interesting use case that I didn't know we supported. They want to store unstructured data, so whatever the JS client sends so no-fixed fields. They send a JSONObject that gets transformed to a string. And they want to return the JSONObject.

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
Merging, but I would like @dsanz input about which parameters are required (pid? portletId?) and which are not.

**Comment by dsanz** [comment]
This pull targets the [active view setting headless api](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/1559168483/Backend+Requirements), which is inspired on the existing active view settings we inherited from the original dataset display.

There, the way to identify the specific active settings is by providing the exact "coordinates" of the dataset display instance id, which are made up by 3 items: the datasetInstance id (provided as the `id` tag attr), the portlet id and the page. All of them are needed as the dataset instance id is not guaranteed to be unique and we want to remember the settings per each datset instance separately.

In [this commit](https://github.com/liferay-frontend/liferay-portal/pull/1182/commits/9c466ed1bb72b963223798d97b7e533f38dbfac8) the original "clayDatasetDisplayId" was renamed to "activeViewId" which may lead to confusion as this new name might suggest to be referring to the resource id. I think that, to respect the original intent, this paremeter should be abstracted out to some other thing with no resemblance to the resource itself (active view settings). A possible alternative would be `componentId`

Does that make sense?

**Comment by nhpatt** [comment]
Yes. I'll integrate the other 2 parameters inside the path after the PR is merged, because required parameters should be path parameters whenever possible.

**Pull request closed without merging**